cmd/cstor-pool-mgmt/controller/replica-controller/handler.go

```diff
@@ -139,6 +109,42 @@ func (c *CStorVolumeReplicaController) cVREventHandler(operation common.QueueOpe
 	return string(apis.CVRStatusInvalid), nil
 }
 
+func (c *CStorVolumeReplicaController) cVRAddEventHandler(cVR *apis.CStorVolumeReplica, fullVolName string) (string, error) {
+	// lock is to synchronize pool and volumereplica. Until certain pool related
+	// operations are over, the volumereplica threads will be held.
+	common.SyncResources.Mux.Lock()
+	if common.SyncResources.IsImported {
+		common.SyncResources.Mux.Unlock()
+		// To check if volume is already imported with pool.
+		importedFlag := common.CheckForInitialImportedPoolVol(common.InitialImportedPoolVol, fullVolName)
+		if importedFlag && !IsInitStatus(cVR) {
+			glog.Infof("CStorVolumeReplica %v is already imported", string(cVR.ObjectMeta.UID))
+			c.recorder.Event(cVR, corev1.EventTypeNormal, string(common.SuccessImported), string(common.MessageResourceImported))
+			return string(apis.CVRStatusOnline), nil
+		}
+	}
+	// If volumereplica is already present.
+	existingvol, _ := volumereplica.GetVolumes()
+	if common.CheckIfPresent(existingvol, fullVolName) {
+		glog.Warningf("CStorVolumeReplica %v is already present", string(cVR.GetUID()))
+		c.recorder.Event(cVR, corev1.EventTypeWarning, string(common.AlreadyPresent), string(common.MessageResourceAlreadyPresent))
+		return string(apis.CVRStatusErrorDuplicate), nil
```

Comment:
Can the err be returned in place of nil?

Reply:
okay, will return fmt.Errorf("Duplicate resource request")

Reply:
Ok